core/templates/app.constants.ts

```diff
@@ -71,6 +73,8 @@ export const AppConstants = {
     INCORRECT_SOLUTION: 'The current solution does not lead to another card.',
     UNRESOLVED_ANSWER:
       'There is an answer among the top 10 which has no explicit feedback.',
+    INVALID_REDIRECTION: `Learner should not be directed back by more` +
```

Comment:
This is still wrong. The whole point of making this dynamic is so that there is no possibility of discrepancy. With the way you've implemented it, it's possible that the `MAX_CARD_COUNT_FOR_VALID_REDIRECTION` could have a different value from what's here. You need to use `MAX_CARD_COUNT_FOR_VALID_REDIRECTION`'s value here.

Reply:
Done.